File: src/com/mishiranu/dashchan/util/AndroidUtils.java
package com.mishiranu.dashchan.util;

import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class AndroidUtils {
	public interface OnReceiveListener {
		void onReceive(BroadcastReceiver receiver, Context context, Intent intent);
	}

	public static BroadcastReceiver createReceiver(OnReceiveListener listener) {
		return new BroadcastReceiver() {
			@Override
			public void onReceive(Context context, Intent intent) {
				listener.onReceive(this, context, intent);
			}
		};
	}

	public static String getApplicationLabel(Context context) {
		return context.getApplicationInfo().loadLabel(context.getPackageManager()).toString();
	}

	public static NotificationChannel createHeadsUpNotificationChannel(String id, CharSequence name) {
		NotificationChannel channel = new NotificationChannel(id, name, NotificationManager.IMPORTANCE_HIGH);
		channel.setSound(null, null);
		channel.setVibrationPattern(new long[] {0});
		return channel;
	}
}

File: src/com/mishiranu/dashchan/util/AndroidUtils.kt
package com.mishiranu.dashchan.util

import android.app.NotificationChannel
import android.app.NotificationManager
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

object AndroidUtils {
    fun interface OnReceiveListener {
        fun onReceive(
            receiver: BroadcastReceiver,
            context: Context,
            intent: Intent,
        )
    }

    @JvmStatic
    fun createReceiver(listener: OnReceiveListener): BroadcastReceiver =
        object : BroadcastReceiver() {
            override fun onReceive(
                context: Context,
                intent: Intent,
            ) {
                listener.onReceive(this, context, intent)
            }
        }

    @JvmStatic
    fun getApplicationLabel(context: Context): String = context.applicationInfo.loadLabel(context.packageManager).toString()

    @JvmStatic
    fun createHeadsUpNotificationChannel(
        id: String,
        name: CharSequence,
    ): NotificationChannel =
        NotificationChannel(id, name, NotificationManager.IMPORTANCE_HIGH).apply {
            setSound(null, null)
            vibrationPattern = longArrayOf(0)
        }
}
